src/bin/taker.rs

```diff
@@ -161,148 +164,139 @@ fn main() -> Result<(), TakerError> {
     };
 
     #[cfg(not(feature = "integration-test"))]
-    let connection_type = ConnectionType::TOR;
+    let mut taker = Taker::init(
+        args.data_directory.clone(),
+        args.wallet_name.clone(),
+        Some(rpc_config.clone()),
+        None,                        // control_port
+        Some(args.tor_auth.clone()), // tor_auth_password
+    )?;
 
     #[cfg(feature = "integration-test")]
-    let connection_type = ConnectionType::CLEARNET;
+    let mut taker = Taker::init(
+        args.data_directory.clone(),
+        args.wallet_name.clone(),
+        Some(rpc_config.clone()),
+        TakerBehavior::Normal,          // behavior
+        None,                           // control_port
+        Some(args.tor_auth.clone()),    // tor_auth_password
+        Some(ConnectionType::CLEARNET), // connection_type
+    )?;
```

Comment:
Instead of this you can just do this

```rust
    let mut taker = Taker::init(
        args.data_directory.clone(),
        args.wallet_name.clone(),
        Some(rpc_config.clone()),
         #[cfg(feature = "integration-test")]
        TakerBehavior::Normal,          // behavior
        None,                           // control_port
        Some(args.tor_auth.clone()), // tor_auth_password
         #[cfg(feature = "integration-test")]    
        Some(ConnectionType::CLEARNET), // connection_type
    )?;
```

You can feature gate individual arguments.